### PR TITLE
Fix problem with provideSequencerStake

### PR DESCRIPTION
**Pull request by AlexChetverov**

### CODE REVIEW

File: test/sequential/maintenance-pallet.api.rolldown.test.ts [hunk 1]
```diff
@@ -240,11 +241,15 @@ describe.each(["mm", "upgradabilityMm"])(
         "%s operations are allowed in mm",
         async (testName) => {
           testLog.getLog().info("DEBUG::TestName - " + testName);
-          let signer: User;
+          let signer: User = getSudoUser();
           const [extrinsic, testsSigner] = tests[testName];
-          if (mmMode === "mm") {
+          if (mmMode === "mm" && testName === "sequencerTearDown") {
```
Comment (AlexChetverov) [inline test/sequential/maintenance-pallet.api.rolldown.test.ts:246]: sequencerTearDown for  "upgradabilityMm" doesn't pass and I don't understand why. Maybe we need to change signer from user to testSigner?

File: test/sequential/sequencerStaking.sequencerJoining.test.ts [hunk 2]
```diff
@@ -14,156 +14,193 @@ import {
...
+    Assets.mintNative(testUser2),
+    Assets.mintNative(testUser3),
+    Assets.mintNative(testUser4),
+    Assets.mintNative(testUser5),
+  );
+  testUser1.addAsset(GASP_ASSET_ID);
+  testUser2.addAsset(GASP_ASSET_ID);
+  testUser3.addAsset(GASP_ASSET_ID);
+  testUser4.addAsset(GASP_ASSET_ID);
+  testUser5.addAsset(GASP_ASSET_ID);
```
Comment (goncer) [inline test/sequential/sequencerStaking.sequencerJoining.test.ts:54]: why this change?

File: test/sequential/sequencerStaking.sequencerJoining.test.ts [hunk 3]
```diff
@@ -14,156 +14,193 @@ import {
...
-    await SequencerStaking.provideSequencerStaking(stakeAmount, chain),
-    testUser.keyRingPair,
+    await SequencerStaking.provideSequencerStaking(
+      testUser5.keyRingPair.address,
+      stakeAmount,
+      chain,
+    ),
+    sudoUser.keyRingPair,
   );
   const eventFiltered = filterAndStringifyFirstEvent(events, "StakeProvided");
```
Comment (goncer) [inline test/sequential/sequencerStaking.sequencerJoining.test.ts:164]: nonce nonce nonce nonce

File: test/sequential/sequencerStaking.sequencerJoining.test.ts [hunk 4]
```diff
@@ -14,156 +14,193 @@ import {
...
   await signTx(
     getApi(),
-    await SequencerStaking.rejoinActiveSequencers(chain),
-    testUser.keyRingPair,
+    await SequencerStaking.rejoinActiveSequencers(
+      chain,
+      testUser4.keyRingPair.address,
+    ),
+    sudoUser.keyRingPair,
   ).then((events) => {
```
Comment (goncer) [inline test/sequential/sequencerStaking.sequencerJoining.test.ts:144]: nonce nonce nonce nonce

File: test/sequential/sequencerStaking.sequencerJoining.test.ts [hunk 5]
```diff
@@ -14,156 +14,193 @@ import {
...
-    await SequencerStaking.provideSequencerStaking(stakeAmount, chain, false),
-    testUser.keyRingPair,
+    await SequencerStaking.provideSequencerStaking(
+      testUser4.keyRingPair.address,
+      stakeAmount,
+      chain,
+      false,
+    ),
+    sudoUser.keyRingPair,
   ).then((events) => {
```
Comment (goncer) [inline test/sequential/sequencerStaking.sequencerJoining.test.ts:128]: nonce nonce nonce

File: test/sequential/sequencerStaking.sequencerJoining.test.ts [hunk 6]
```diff
@@ -14,156 +14,193 @@ import {
...
     getApi(),
-    await SequencerStaking.provideSequencerStaking(stakeAmount, chain, true),
-    testUser.keyRingPair,
+    await SequencerStaking.provideSequencerStaking(
+      testUser3.keyRingPair.address,
+      stakeAmount,
+      chain,
+      true,
+    ),
+    sudoUser.keyRingPair,
```
Comment (goncer) [inline test/sequential/sequencerStaking.sequencerJoining.test.ts:107]: nonce nonce nonce

File: test/sequential/sequencerStaking.sequencerJoining.test.ts [hunk 7]
```diff
@@ -14,156 +14,193 @@ import {
...
       (await SequencerStaking.minimalStakeAmount()).subn(1000),
       chain,
       true,
     ),
-    testUser.keyRingPair,
-  ).then((events) => {
-    const eventResponse = getEventResultFromMangataTx(events);
-    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
-    expect(eventResponse.data).toEqual("NotEnoughSequencerStake");
+    sudoUser.keyRingPair,
```
Comment (goncer) [inline test/sequential/sequencerStaking.sequencerJoining.test.ts:88]: nonce

File: test/sequential/sequencerStaking.sequencerJoining.test.ts [hunk 8]
```diff
@@ -14,156 +14,193 @@ import {
...
     getApi(),
-    await SequencerStaking.provideSequencerStaking(stakeAmount, chain, false),
-    testUser.keyRingPair,
+    await SequencerStaking.provideSequencerStaking(
+      testUser1.keyRingPair.address,
+      stakeAmount,
+      chain,
+      false,
+    ),
+    sudoUser.keyRingPair,
```
Comment (goncer) [inline test/sequential/sequencerStaking.sequencerJoining.test.ts:66]: nonce

File: test/sequential/maintenance-pallet.api.rolldown.test.ts [hunk 9]
```diff
@@ -240,11 +241,15 @@ describe.each(["mm", "upgradabilityMm"])(
...
             signer = testsSigner;
-          } else {
+          }
+          if (
+            mmMode === "upgradabilityMm" &&
+            testName === "sequencerTearDown"
+          ) {
```
Comment (goncer) [inline test/sequential/maintenance-pallet.api.rolldown.test.ts:252]: ???

File: test/sequential/maintenance-pallet.api.rolldown.test.ts (hunk 1 above)
Comment (goncer) [reply test/sequential/maintenance-pallet.api.rolldown.test.ts:246]: I'll think on something else,

**Review by goncer — CHANGES_REQUESTED** [review]
looking at the mm test

File: test/sequential/sequencerStaking.sequencerJoining.test.ts (hunk 2 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.sequencerJoining.test.ts:54]: I received a blinking error when I was working with these tests. They worked well separately, but 2 tests had Tx Map errors when I ran them together.

File: test/sequential/sequencerStaking.sequencerJoining.test.ts (hunk 8 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.sequencerJoining.test.ts:66]: add

File: test/sequential/sequencerStaking.sequencerJoining.test.ts (hunk 7 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.sequencerJoining.test.ts:88]: add

File: test/sequential/sequencerStaking.sequencerJoining.test.ts (hunk 6 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.sequencerJoining.test.ts:107]: add

File: test/sequential/sequencerStaking.sequencerJoining.test.ts (hunk 5 above)
Comment (AlexChetverov) [reply test/sequential/sequencerStaking.sequencerJoining.test.ts:128]: add
